models/repo/pushmirror.go

```diff
@@ -64,18 +64,26 @@ func UpdatePushMirror(m *PushMirror) error {
 }
 
 // DeletePushMirrorByID deletes a push-mirrors by ID
+// WARNING: This does not check if this PushMirror belongs to a RepoID
 func DeletePushMirrorByID(ID int64) error {
 	_, err := db.GetEngine(db.DefaultContext).ID(ID).Delete(&PushMirror{})
 	return err
 }
 
+// DeletePushMirrorByRepoIDAndName deletes a push-mirrors by remote name
+func DeletePushMirrorByRepoIDAndName(repoID int64, remoteName string) error {
+	_, err := db.GetEngine(db.DefaultContext).Where("repo_id = ? AND remote_name = ?", repoID, remoteName).Delete(&PushMirror{})
+	return err
+}
+
 // DeletePushMirrorsByRepoID deletes all push-mirrors by repoID
 func DeletePushMirrorsByRepoID(repoID int64) error {
 	_, err := db.GetEngine(db.DefaultContext).Delete(&PushMirror{RepoID: repoID})
```

Comment:
It's a chance to merge the 3 functions as 1

Reply:
from my point of view, these are three distinct functions. I don't know if there is any benefit in merging them.

Reply:
maybe we can merge the two functions `DeletePushMirrorByID(ID int64)` and `DeletePushMirrorByRepoIDAndName(repoID int64, remoteName string)` 
 
into 

```go
func DeletePushMirror(pushMirror PushMirror) error {
	if pushMirror.RepoID > 0 {
		//delete using remoteName
		if pushMirror.RemoteName != "" {
			_, err := db.GetEngine(db.DefaultContext).Where("repo_id = ? AND remote_name = ?", pushMirror.RepoID, pushMirror.RemoteName).Delete(&PushMirror{})
			return err
		}
		// delete using ID
		if pushMirror.ID > 0 {
			_, err := db.GetEngine(db.DefaultContext).Where("repo_id = ? AND id = ?", pushMirror.RepoID, pushMirror.ID).Delete(&PushMirror{})
			return err
		}
		return errors.New("PushMirror ID or RemoteName required")
	} else {
		return errors.New("repoID required and must be set")
	}
}
```

but for `DeletePushMirrorsByRepoID(repoID int64)` for safety it will be better to keep it as a separate function

Reply:
It's better to have a function named `func DeletePushMirrors(opts DeletePushMirrorOptions{
RepoID:
RemoteName:
ID:
})

Reply:
Or return an error like `you needs at least one condition`.